include/kframework/evm-types.md

```diff
@@ -360,16 +359,16 @@ Bytes helper functions
     syntax Bytes ::= #range ( Bytes , Int , Int ) [function, total]
  // ---------------------------------------------------------------
     rule                #range(_, START, WIDTH)  => .Bytes                                                                       requires notBool (WIDTH >=Int 0 andBool START >=Int 0)
-    rule [bytesRange] : #range(WS, START, WIDTH) => substrBytes(padRightBytes(WS, START +Int WIDTH, 0), START, START +Int WIDTH) requires WIDTH >=Int 0 andBool START >=Int 0 andBool START <Int lengthBytes(WS)
+    rule [bytesRange] : #range(WS, START, WIDTH) => substrBytes(padRightBytes(WS, START +Int WIDTH, 0), START, START +Int WIDTH) requires WIDTH >=Int 0 andBool START >=Int 0 andBool START <Int lengthBytes(WS) [preserves-definedness]
```

Comment:
`#range` is total.... I guess since `#range` is total, it should be already preserving definedness? That should be derivable from the axioms of total functions I think. Basically, "for every single-element input, it will return a single value" (totality, functionality), and the variables here are all element variables (not set variables). So I think we can just make the definedness checker stronger. Currently we only inspect the RHS and REQ clause to see if they are all total functions and constructors, but another sufficient criteria could be: The LHS only contains total functions, constructors, domain values, and element variables. Is taht true?

Reply:
The `total` attribute to a function is currently not considered to infer definedness-preservation. 
I agree we could modify the analysis to just assume a rule is preserving definedness  when it uses a total function on the LHS.
Currently the rule won't be considered definedness-preserving since `padRightBytes` , a partial function, is used on the RHS. So the `preserves-definedness` is re-stating the totality of `#range`, in a way.